tests/integration/test_network_plugin.py

```diff
@@ -40,15 +39,33 @@ def setUp(self):
     def tearDown(self):
         super(NatRulesOperationsTestCase, self).tearDown()
 
-    def test_nat_rules_create_delete(self):
-        self.assertNotIn(self.public_ip, collectExternalIps(
-            self._get_gateway()))
+
+    def test_nat_rules_create_delete_with_explicit_ip(self):
+        self.ctx.node.properties['floatingip'].clear()
```

Comment:
I think this line is unnecessary - you're recreating self.ctx before every test in setUp